### PR TITLE
Trammel branch

### PR DESCRIPTION
**Pull request by tramstheman**

### CODE REVIEW

File: src/com/northvine/reader/implementation/DataMatrix.java [hunk 1]
```diff
@@ -111,7 +147,34 @@ public boolean readText(final String text)
...
+            else
+            {
+               generated[8 - bit] += BLACK_CHAR;
+               asciiCode %= (1<<bit);
+            }
+         }
+         // Add bottom Closed Limitation Line
+         generated[9] += BLACK_CHAR;
+      }
+      return scan(new BarcodeImage(generated)); // finished generating image
```
Comment (tramstheman) [inline src/com/northvine/reader/implementation/DataMatrix.java]: Now scan is called, so by the time we want to print to console, the correct actual width and actual height should be golden

File: src/com/northvine/reader/implementation/DataMatrix.java [hunk 2]
```diff
@@ -137,110 +213,186 @@ public boolean translateImageToText()
...
+            System.out.println(String.format("X: %s, Y: %s, Value: %s", x, y, barcodeImage.getPixel(y, x)));
+            if(barcodeImage.getPixel(y, x))
             {
-               return row;
+               return y;
             }
          }
       }
-      return 0;
+      throw new RuntimeException("The image state leaves vertical scanning undetectable");
```
Comment (tramstheman) [inline src/com/northvine/reader/implementation/DataMatrix.java]: this keeps getting thrown, notice I am also printing out each x and y value, and it is reading everything as false.

**Comment by GEvanscsumb** [comment]
looks good to me, will wait for other to checkout my additions to the PR.

File: src/com/northvine/reader/implementation/DataMatrix.java [hunk 3]
```diff
@@ -106,13 +105,11 @@ public boolean scan(final BarcodeImage barcodeImage)
...
       {
          this.barcodeImage = barcodeImage.clone();
-
          cleanImage();
          actualWidth = computeSignalWidth();
          actualHeight = computeSignalHeight();
          return true;
-      }
-      catch(CloneNotSupportedException e)
+      } catch (CloneNotSupportedException e)
```
Comment (tramstheman) [inline src/com/northvine/reader/implementation/DataMatrix.java]: The catch should be on its own line

File: src/com/northvine/reader/implementation/DataMatrix.java [hunk 4]
```diff
@@ -122,34 +119,40 @@ public boolean scan(final BarcodeImage barcodeImage)
...
+    * @param text
+    *           represents the text string that is desired to be encoded in the
+    *           image.
     * @return a boolean representing the success of this mutation
     */
    @Override
    public boolean readText(final String text)
    {
-      this.text = text + "ª";
+      this.text = text;
```
Comment (tramstheman) [inline src/com/northvine/reader/implementation/DataMatrix.java]: "ª" is the delimiter for the QR code, see the example

**Comment by GEvanscsumb** [comment]
@tramstheman makes sense, my apologies.

**Comment by tramstheman** [comment]
No worries! Always a learning experience

**Comment by KyleLuoma** [comment]
Sorry, I committed the clone() patch directly to your branch Trammel. I meant to do a PR...

**Pull request merged**